src/Cli/dotnet/commands/dotnet-workload/install/MsiInstallerBase.cs

```diff
@@ -272,6 +275,40 @@ protected uint RepairMsi(string productCode, string logFile)
             throw new InvalidOperationException($"Invalid configuration: elevated: {IsElevated}, client: {IsClient}");
         }
 
+        /// <summary>
+        /// Instructs future workload operations to use workload sets or loose manifests, per newMode.
+        /// </summary>
+        /// <param name="sdkFeatureBand">The feature band to update</param>
+        /// <param name="newMode">Where to use loose manifests or workload sets</param>
+        /// <param name="logFile">Full path of the log file</param>
+        /// <returns>Error code indicating the result of the operation</returns>
+        /// <exception cref="InvalidOperationException"></exception>
+        protected void UpdateInstallMode(SdkFeatureBand sdkFeatureBand, bool newMode)
+        {
+            Elevate();
+
+            if (IsElevated)
+            {
+                string path = Path.Combine(WorkloadInstallType.GetInstallStateFolder(sdkFeatureBand, DotNetHome), "default.json");
+                // Create the parent folder for the state file and set up all required ACLs
+                SecurityUtils.CreateSecureDirectory(Path.GetDirectoryName(path));
+                var installStateContents = InstallStateContents.FromString(File.Exists(path) ? File.ReadAllText(path) : "{}");
```

Comment:
Same as the file based installer, I'd suggest creating an `InstallStateContents` object directly instead of parsing JSON if the file doesn't exist.